src/storage/predicate.rs

```diff
@@ -0,0 +1,135 @@
+use std::borrow::Cow;
+
+use crate::error::InterpreterError;
+use crate::storage::InterpreterStorage;
+
+use fuel_asm::Word;
+use fuel_storage::{MerkleRoot, MerkleStorage, Storage};
+use fuel_tx::Contract;
+use fuel_types::{Address, AssetId, Bytes32, ContractId, Salt};
+
+/// No-op storage used for predicate operations.
+///
+/// The storage implementations are expected to provide KV-like operations for contract operations.
+/// However, predicates, as defined in the protocol, cannot execute contract opcodes. This means
+/// its storage backend for predicate execution shouldn't provide any functionality.
+#[derive(Debug, Default, Clone, Copy)]
+pub struct PredicateStorage;
+
+impl Storage<ContractId, Contract> for PredicateStorage {
```

Comment:
I think the only way to achieve this would be to partition our main opcode executor match into a match for opcodes that use storage and a separate match over opcodes that don't have storage. Doubling matching every opcode would incur a higher runtime penalty.

Once we have the bitfield to determine which opcodes are storage related, we could avoid running both matches and select the correct opcode matcher.